modules/nlu/src/views/full/entities/PatternEntity.tsx

```diff
@@ -0,0 +1,149 @@
+import {
+  Checkbox,
+  Colors,
+  FormGroup,
+  H1,
+  Icon,
+  InputGroup,
+  Label,
+  Position,
+  Tag,
+  TextArea,
+  Tooltip
+} from '@blueprintjs/core'
+import { NLU } from 'botpress/sdk'
+import _ from 'lodash'
+import React, { useEffect, useState } from 'react'
+
+import style from './style.scss'
+
+interface Props {
+  entity: NLU.EntityDefinition
+  updateEntity: (entity: NLU.EntityDefinition) => void
+}
+
+export const PatternEntityEditor: React.FC<Props> = props => {
+  const [matchCase, setMatchCase] = useState<boolean>(props.entity.matchCase)
+  const [sensitive, setSensitive] = useState<boolean>(props.entity.sensitive)
+  const [pattern, setPattern] = useState<string>(props.entity.pattern)
+  const [paternValid, setPatternValid] = useState<boolean>(true)
+  const [examplesStr, setExampleStr] = useState((props.entity.examples || []).join('\n'))
+  const [allExamplesMatch, setExamplesMatch] = useState<boolean>(true)
+
+  const validateExamples = _.debounce(() => {
+    let p = pattern
+    if (!p.startsWith('^')) {
+      p = '^' + p
+    }
+    if (!p.endsWith('$')) {
+      p = p + '$'
+    }
+    const rx = new RegExp(p, matchCase ? '' : 'i')
+    const allMatching = examplesStr
+      .split('\n')
+      .filter(ex => ex !== '')
+      .map(ex => rx.test(ex))
+      .every(Boolean)
+
+    setExamplesMatch(allMatching)
+  }, 750)
+
+  useEffect(() => {
+    try {
+      new RegExp(pattern)
+      setPatternValid(true)
+      const newEntity: NLU.EntityDefinition = {
+        ...props.entity,
+        pattern,
+        sensitive,
+        matchCase,
+        examples: examplesStr.trim().split('\n')
+      }
+      validateExamples()
+      props.updateEntity(newEntity)
+    } catch (e) {
+      setPatternValid(false)
+    }
+  }, [pattern, matchCase, sensitive, examplesStr])
+
+  return (
+    <div className={style.entityEditorBody}>
+      <div className={style.dataPane}>
+        <FormGroup
+          label="Regular expression"
+          labelFor="pattern"
+          labelInfo={
+            paternValid ? null : (
+              <Tag intent="danger" minimal className={style.validationTag}>
+                Invalid pattern
+              </Tag>
+            )
+          }
+        >
+          <InputGroup
+            leftIcon={<Icon iconSize={20} className={style.regexInputDash} icon="slash" />}
+            rightElement={<Icon iconSize={20} className={style.regexInputDash} icon="slash" />}
+            type="text"
+            id="pattern"
+            placeholder="Insert a valid pattern"
+            value={pattern}
+            intent={paternValid ? 'none' : 'danger'}
+            onChange={e => setPattern(e.target.value)}
+          />
+        </FormGroup>
+        <FormGroup
+          label="Matching examples"
+          labelFor="examples"
+          labelInfo={
+            examplesStr &&
```

Comment:
I'd add a warning to tell people they **need** to provide examples as they will become mandatory in future versions of Botpress (to train their slots)